pkg/controller/dynamic/dynamic_controller_integration_test.go

```
@@ -774,6 +776,46 @@ func containsResourceIDTestVar(t *testing.T, u *unstructured.Unstructured) bool
 	return strings.Contains(string(b), resourceIDTestVar)
 }
 
+func verifyStatusFields(t *testing.T, reconciledUnstruct *unstructured.Unstructured) {
+	statusUnstruct, ok := reconciledUnstruct.Object["status"]
+	if !ok {
+		t.Errorf("reconciled resource didn't have 'status' field but it should")
+	}
+	statusMap, ok := statusUnstruct.(map[string]interface{})
+	if !ok {
+		t.Errorf("reconciled status didn't have type map[string]interface{} but was actually %T", statusUnstruct)
+	}
+
+	requiredFieldsMap := map[string]bool{"observedGeneration": true, "conditions": true}
+	if len(statusMap) < 2 {
+		t.Errorf("reconciled resource didn't have enough subfields under 'status' field: it should at least have fields 'conditions' and 'observedGeneration'")
+	}
+
+	if !k8s.OutputOnlyFieldsAreUnderObservedState(reconciledUnstruct.GroupVersionKind()) {
+		return
```

Do we want to return an error here?